extending-valtimo/access-control/registering-a-resource.md

```diff
@@ -0,0 +1,104 @@
+# Registering a resource
+
+Access control can be exercised over resources that have been registered. Valtimo registers a few resources out of 
+the box. These are:
+- `com.ritense.document.domain.impl.JsonSchemaDocument`
+- `com.ritense.document.domain.impl.searchfield.SearchField`
+- `com.ritense.note.domain.Note`
+- `com.ritense.valtimo.camunda.domain.CamundaTask`
+
+The authorization module allows developers to register their own resources, which allows said resources to be used when
+configuring permissions.
+
+## Registering a resource
+
+In order to register a resource for use in access control, there are two interfaces that should be implemented; the
+`AuthorizationSpecification`, and the `AuthorizationSpecificationFactory`.
+
+### Specification
+The `AuthorizationSpecification` is an abstract class built upon the Spring `Specification` interface. Generally, only
+two methods have to be overridden. The `toPredicate` method creates predicates that are used to enhance queries. For
+example, when a user looks at a list of documents, only the documents this user has access to will be shown.
+
+Secondly, the `identifierToEntity` method is used to go from an identifier to the actual entity. 
```

Comment:
I think a bit more explanation on how this is used would be welcome

Reply:
added some more details